.github/container/test-t5x-fmha.sh

```diff
@@ -0,0 +1,244 @@
+    echo "  -d, --dtype               Data type, defaults to bfloat16."
+    echo "  --enable-te {0,1}         1 to enable, 0 to disable; defaults to ENABLE_TE in env or 0 if unset"
+    echo "  -e, --epochs              Number of epochs to run, defaults to 7."
+    echo "  --multiprocess            Enable the multiprocess GPU mode."
+    echo "  -o, --output NAME         Name for the output folder, a temporary folder will be created if none specified."
+    echo "  --save-hlo {0, 1}         1 to save the dumped hlo, 0 to remove the hlo dumped folder"
+    echo "  --seed INT                Random seed for deterministim. Defaults to 42."
+    echo "  -s, --steps-per-epoch INT Steps per epoch. Detauls to 100"
+    echo "  -h, --help                Print usage."
+    exit $1
+}
+
+args=$(getopt -o a:b:cd:e:ho:s: --long additional-args:,batch-size:,use-contrib-configs,dtype:,enable-te:,epochs:,help,multiprocess,output:,seed:,steps-per-epoch: -- "$@")
+if [[ $? -ne 0 ]]; then
+    exit 1
+fi
+
+# Default arguments
+
+ADDITIONAL_ARGS=""
+BATCH_SIZE=0
+USE_CONTRIB_CONFIGS=0
+DTYPE=bfloat16
+EPOCHS=7
+MULTIPROCESS=0
+OUTPUT=$(mktemp -d)
+SEED=42
+STEPS_PER_EPOCH=100
+ENABLE_TE=${ENABLE_TE:-0}
+SAVE_HLO=${SAVE_HLO:-1}
+
+eval set -- "$args"
+while [ : ]; do
+    case "$1" in
+        -a | --additional-args)
+            ADDITIONAL_ARGS="$2"
+            shift 2
+            ;;
+        -b | --batch-size)
+            BATCH_SIZE="$2"
+            shift 2
+            ;;
+        -c | --use-contrib-configs)
+            USE_CONTRIB_CONFIGS=1
+            shift 1
+            ;;
+        -d | --dtype)
+            DTYPE="$2"
+            shift 2
+            ;;
+        --enable-te)
+            ENABLE_TE="$2"
+            shift 2
+            ;;
+        -e | --epochs)
+            EPOCHS="$2"
+            shift 2
+            ;;
+        -h | --help)
+            usage 1
+            ;;
+        --multiprocess)
+            MULTIPROCESS=1
+            shift 1
+            ;;
+        -o | --output)
+            OUTPUT="$2"
+            shift 2
+            ;;
+        --save-hlo)
+            SAVE_HLO="$2"
+            shift 2
+            ;;
+        --seed)
+            SEED="$2"
+            shift 2
+            ;;
+        -s | --steps-per-epoch)
+            STEPS_PER_EPOCH="$2"
+            shift 2
+            ;;
+        --)
+            shift;
+            break 
+            ;;
+        *)
+            echo "UNKNOWN OPTION $1"
+            usage 1
+    esac
+done
+
+if [[ $BATCH_SIZE == 0 ]]; then
+    echo "ERROR: Batch size must be specified."
+    usage 1
+fi
+
+# Set hlo dump folder after output folder is set.
+HLO_DIR=${OUTPUT}/hlo
+
+## Setting the env variables for FMHA
+export XLA_FLAGS="--xla_dump_hlo_as_text --xla_dump_to=${HLO_DIR} --xla_gpu_graph_level=0 --xla_gpu_fused_attention_use_cudnn_rng=true"
```

Comment:
So these flags will override anything in the container, which could possibly be needed. I prefer this pattern for setting these env vars:
```sh
export BASE_XLA_FLAGS="${BASE_XLA_FLAGS:---xla_dump_hlo_as_text --xla_dump_to=${HLO_DIR} --xla_gpu_graph_level=0 --xla_gpu_fused_attention_use_cudnn_rng=true}"
export XLA_FLAGS="${BASE_XLA_FLAGS} ${XLA_FLAGS:-}"
```

That way you pick up the environment, but you can still "zero out" the env var in this script without a code base by just doing `BASE_XLA_FLAGS=""`

Reply:
Incorporated review comments.